opa/opa.go

```diff
@@ -51,51 +62,47 @@ func (v *PolicyViolation) Write(w io.Writer) {
 }
 
 // NewClient OpaClient constructor
-func NewClient(logger *zap.Logger, host string) *Client {
-	client := &Client{
+func NewClient(logger *zap.Logger, host string) Client {
+	client := &client{
 		logger: logger,
 		Host:   host,
 	}
 	return client
 }
 
 // InitializePolicy initializes OPA policy if it does not already exist
-func (opa *Client) InitializePolicy(policy string) ClientError {
+func (opa *client) InitializePolicy(policy string) ClientError {
 	_ = opa.logger.Named("Initialize Policy")
 
 	exists, err := opa.policyExists(policy)
 	if err != nil {
-		return clientError{"error checking if policy exists", OpaClientErrorTypePolicyExists, err}
+		return NewClientError("error checking if policy exists", OpaClientErrorTypeGetPolicy, err)
 	}
 
 	if !exists {
-		// fetch violations from ES
-		violations := []PolicyViolation{}
-		err = opa.publishPolicy(policy, violations)
-		if err != nil {
-			return clientError{"error publishing policy", OpaClientErrorTypePublishPolicy, err}
-		}
+		// TODO: fetch rules from ES
+		return NewClientError("policy does not exist", OpaClientErrorTypePolicyNotFound, nil)
 	}
 
 	return nil
 }
 
 // EvaluatePolicy evalutes OPA policy agains provided input
-func (opa *Client) EvaluatePolicy(policy string, input string) (*EvaluatePolicyResult, error) {
+func (opa *client) EvaluatePolicy(policy string, input []byte) (*EvaluatePolicyResponse, error) {
 	log := opa.logger.Named("Evalute Policy")
 	request, err := json.Marshal(&EvalutePolicyRequest{Input: json.RawMessage(input)})
 	if err != nil {
-		log.Error("failed to encode OPA input", zap.Error(err), zap.String("input", input))
+		log.Error("failed to encode OPA input", zap.Error(err), zap.String("input", string(input)))
 		return nil, fmt.Errorf("failed to encode OPA input: %s", err)
 	}
-	httpResponse, err := http.Post(opa.getURL(fmt.Sprintf("v1/data/%s", policy)), "application/json", bytes.NewReader(request))
+	httpResponse, err := http.Post(opa.getURL(fmt.Sprintf("v1/data/%s?explain=full&pretty", policy)), "application/json", bytes.NewReader(request))
```

Comment:
is the `explain=full&pretty` needed?  or was that just used for testing?

Reply:
It is needed for debugging. I added a flag to the OPA client so that it can be enabled/disabled